src/System.Collections/tests/Performance/Perf.Dictionary.cs

```diff
@@ -170,28 +170,339 @@ public void TryGetValue(int size)
+                sampleValues[i] = i;
+
+                dictionary.Add(sampleKeys[i], sampleValues[i]);
+            }
+
+            bool result = false;
+
+            int missingValue = sampleLength;   //The value sampleLength is not present in the dictionary.
+
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    for (int j = 0; j < sampleLength; j++)
+                        result = dictionary.ContainsValue(missingValue);
+        }
+
+        [Benchmark]
+        [InlineData(1000)]
+        [InlineData(10000)]
+        [InlineData(100000)]
+        public static void Dictionary_ContainsValue_String_True(int sampleLength)
+        {
+            string[] sampleKeys = new string[sampleLength];
+            string[] sampleValues = new string[sampleLength];
+
+            Dictionary<string, string> dictionary = new Dictionary<string, string>();
+
+            for (int i = 0; i < sampleLength; i++)
+            {
+                sampleKeys[i] = i.ToString();
+                sampleValues[i] = i.ToString();
+
+                dictionary.Add(sampleKeys[i], sampleValues[i]);
+            }
+
+            bool result = false;
+
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    for (int j = 0; j < sampleLength; j++)
+                        result = dictionary.ContainsValue(sampleValues[j]);   //Every value searched for is present in the dictionary.
+        }
+
+        [Benchmark]
+        [InlineData(1000)]
+        [InlineData(10000)]
+        [InlineData(100000)]
+        public static void Dictionary_ContainsValue_String_False(int sampleLength)
+        {
+            string[] sampleKeys = new string[sampleLength];
+            string[] sampleValues = new string[sampleLength];
+
+            Dictionary<string, string> dictionary = new Dictionary<string, string>();
+
+            for (int i = 0; i < sampleLength; i++)
+            {
+                sampleKeys[i] = i.ToString();
+                sampleValues[i] = i.ToString();
+
+                dictionary.Add(sampleKeys[i], sampleValues[i]);
+            }
+
+            bool result = false;
+
+            string missingValue = sampleLength.ToString();   //The string representation of sampleLength is not present in the dictionary.
+
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    for (int j = 0; j < sampleLength; j++)
+                        result = dictionary.ContainsValue(missingValue);
+        }
+
+
+        [Benchmark]
+        [InlineData(1000)]
+        [InlineData(10000)]
+        [InlineData(100000)]
+        public static void Dictionary_ContainsKey_Int_True(int sampleLength)
+        {
+            int[] sampleKeys = new int[sampleLength];
+            int[] sampleValues = new int[sampleLength];
```

Comment:
Is this sampleValues array actually necessary?  Seems like the for loop below could just write i into the dictionary.  Similarly since the keys are just the values 0 through sampleLength, sampleKeys shouldn't be needed either, and we could avoid those lookups as part of the perf loop.